core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -628,6 +630,22 @@ private[spark] class Executor(
           }
       } finally {
         runningTasks.remove(taskId)
+        if(idleGCEnabled) {
+          if (runningTasks.isEmpty) {
```

Comment:
Is this enough to check whether executor is idle or not? I'm afraid the follow-up stages will come soon, but due to the full-gc, it makes the current executors unavailable.

There're plenty of ways to trigger GC manually if you do think this is proper/idle time for executor to do full-gc, like jconsole, jvisualvm. Also real-world workloads are quite different. IMHO putting this mechanism in executor automatically seems a little strange to me.

Reply:
Will only call GC when heap usage is more than 70 or 80% used. heap gets cleared only when context cleaner run's full GC(which run's for every 30 minutes). in between the executor is idle, why don't we use that time for GC?

Reply:
I think that the concern is that the executor doesn't know when it will receive another task. Consider the case where an executor has 1 core and high memory usage due to broadcast tables. This would trigger a GC between each task, which could delay the entire job.

I think it would be better to coordinate with the driver, which knows whether there are more tasks in a stage (or if there are multiple concurrent stages). Seems to me that a stage boundary is the right time to optimistically trigger GC. Of course, that only works if the executor is going to keep running.

Reply:
Yes, that's my concern. Executor doesn't have lots of information to check whether it will be idle for how long time. The current way of checking seems not so good.